src/ffs/pub/tree.ts

```diff
@@ -0,0 +1,78 @@
+import pathUtil from '../path'
+import publicNode, { PublicNode } from "./node"
+import { Tree, Link } from "../types"
+import { CID, FileContent } from '../../ipfs'
+import { getRecurse, addChildRecurse } from "../helpers"
+
+export class PublicTree implements Tree {
+
+  root: PublicNode
+
+  constructor(root: PublicNode) {
+    this.root = root
+  }
+
+  async put(): Promise<CID> {
+    return this.root.put()
+  }
+
+  async cid(): Promise<CID> {
+    return this.put()
+  }
+
+  async listDir(path: string): Promise<Link[] | null> {
+    const node = await this.getNode(path)
+    return node?.links() || []
+  }
+
+  async makeDir(path: string): Promise<PublicTree> {
+    const toAdd = await publicNode.empty()
```

Comment:
My editor is warning that the `await` doesn't do anything here

Reply:
So this is a weird one because `PrivateTree.mkdir` is async but `PublicTree.mkdir` is not. But I think we want to keep the API the same :thinking:

Reply:
Oh I see. Yeah, okay that makes sense. Better than switching on each. I wonder if we can suppress that warning?